app/models/status.rb

```diff
@@ -79,7 +79,9 @@ class Status < ApplicationRecord
   has_many :local_bookmarked, -> { merge(Account.local) }, through: :bookmarks, source: :account
 
   has_and_belongs_to_many :tags
-  has_and_belongs_to_many :preview_cards
+
+  has_many :preview_cards_statuses, dependent: :destroy
+  has_many :preview_cards, -> { select('preview_cards.*, preview_cards_statuses.url AS original_url').joins(:preview_cards_statuses) }, through: :preview_cards_statuses
```

Comment:
Although I understand this is done to have minimal changes to the codebase, I'm uncomfortable with the model of a given record being populated with different attributes depending on where it was loaded from. Conceptually, this doesn't look very clean, and seems very fragile.

I'm not sure whether we should try going this way, or explicitly reference the model for the join table wherever we actually use it.